dspace/modules/xmlui/src/main/java/org/dspace/app/xmlui/aspect/versioning/VersionManager.java

```diff
@@ -0,0 +1,199 @@
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.WorkspaceItemService;
+import org.dspace.core.Context;
+import org.dspace.versioning.Version;
+import org.dspace.versioning.VersionHistory;
+import org.dspace.versioning.factory.VersionServiceFactory;
+import org.dspace.versioning.service.VersionHistoryService;
+import org.dspace.versioning.service.VersioningService;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.UUID;
+
+/**
+ *
+ *
+ * @author Fabio Bolognesi (fabio at atmire dot com)
+ * @author Mark Diggory (markd at atmire dot com)
+ * @author Ben Bosman (ben at atmire dot com)
+ */
+public class VersionManager {
+
+    private static final Message T_version_created = new Message("default", "The new version has been created.");
+    private static final Message T_version_delete = new Message("default", "The selected version(s) have been deleted.");
+    private static final Message T_version_updated = new Message("default", "The version has been updated.");
+    private static final Message T_version_restored = new Message("default", "The version has been restored.");
+
+    protected static final AuthorizeService authorizeService = AuthorizeServiceFactory.getInstance().getAuthorizeService();
+    protected static final ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+    protected static final VersionHistoryService versionHistoryService = VersionServiceFactory.getInstance().getVersionHistoryService();
+    protected static final VersioningService versioningService = VersionServiceFactory.getInstance().getVersionService();
+    protected static final WorkspaceItemService workspaceItemService = ContentServiceFactory.getInstance().getWorkspaceItemService();
+
+
+    /**
+     * Create a new version of the specified item
+     *
+     * @param context The DSpace context
+     * @param itemID  The id of the to-be-versioned item
+     * @param summary summary.
+     * @return A result object
+     * @throws java.sql.SQLException passed through.
+     * @throws org.dspace.authorize.AuthorizeException passed through.
+     * @throws java.io.IOException passed through.
+     */
+    public static FlowResult processCreateNewVersion(Context context, UUID itemID, String summary)
+            throws SQLException, AuthorizeException, IOException {
+        FlowResult result = new FlowResult();
+        try {
+            result.setContinue(false);
+
+            Item item = itemService.find(context, itemID);
+
+            if (authorizeService.authorizeVersioning(context, item) || itemService.canEdit(context, item)) {
+                Version version = versioningService.createNewVersion(context, item, summary);
+                WorkspaceItem wsi = workspaceItemService.findByItem(context, version.getItem());
+                
+                // Force data to be written to the database
+                context.commit();
```

Comment:
`context.commit()` is needed to persist the newly created Version and associated WorkspaceItem in order to be retrieved in the cocoon process workflow.

This fix is not in the 6.0 Release, https://github.com/DSpace/DSpace/releases/tag/dspace-6.0. It is however in the latest branch for dspace-6_x, https://github.com/DSpace/DSpace/blob/dspace-6_x/dspace-xmlui/src/main/java/org/dspace/app/xmlui/aspect/versioning/VersionManager.java.